Source/Meadow.Core/Cloud/MeadowCloudConnectionService.cs

```diff
@@ -310,7 +308,7 @@ private async void ConnectionStateMachine()
         // update state machine
         try
         {
-            while (!_stopService)
+            while (!_stopService && !Resolver.App.CancellationToken.IsCancellationRequested)
```

Comment:
I think we don't want that. I have also tried to keep the update service running past the app. Eventually this allows us to hold off on restart until Meadow.Cloud gets a chance to receive a OTA update that fixes the stopped/cancelled app.